src/components/forms/TextField/TextField.stories.ts

```diff
@@ -141,3 +141,113 @@ export const ValidateOnBlur: Story = {
+        showCharCount: '',
+      },
+    },
+  },
+  globals: {
+    locale: 'nl',
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+    const input = await canvas.findByLabelText('Show char count');
+
+    await userEvent.type(input, 'I am an example text.');
+
+    // Expect the 'X characters' text to be shown, with the right amount of used
+    // characters.
+    const charactersUsedMessage = await canvas.findByText('21 karakters.');
+    expect(charactersUsedMessage).toBeVisible();
+
+    // Expect that the character count is also accessible by screen-readers.
+    await waitFor(() => {
+      expect(input).toHaveAccessibleDescription('21 karakters.');
+    });
+  },
+};
+
+export const ShowCharCountWithLimit: Story = {
+  name: 'Show character count with a character limit',
+  args: {
+    name: 'showCharCount',
+    label: 'Show char count',
+    showCharCount: true,
+    maxLength: 50,
+  },
+  parameters: {
+    formik: {
+      initialValues: {
+        showCharCount: '',
+      },
+      zodSchema: z.object({
+        showCharCount: z.string().max(50),
+      }),
+    },
+  },
+  globals: {
+    locale: 'nl',
+  },
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+    const input = await canvas.findByLabelText('Show char count');
+
+    // The character limit should not be set to the html component.
+    // Conform the current setup, we limit the amount of characters via custom validation
+    expect(input).not.toHaveAttribute('maxlength', '50');
+
+    await step('Character count is shown after typing content', async () => {
+      await userEvent.type(input, 'I am an example text.');
+
+      // Expect the 'characters remaining' text to be shown, with the right amount of
+      // remaining characters.
+      expect(await canvas.findByText('Nog 29 karakters over.')).toBeVisible();
+      await waitFor(() => {
+        expect(input).toHaveAccessibleDescription('Nog 29 karakters over.');
+      });
+    });
+
+    await step('Character count for 1 character remaining', async () => {
+      // Show the correct message when 1 character left.
+      await userEvent.type(input, " I'm standing on the edge...");
+
+      expect(await canvas.findByText('Nog 1 karakter over.')).toBeVisible();
+      await waitFor(() => {
+        expect(input).toHaveAccessibleDescription('Nog 1 karakter over.');
+      });
+    });
+
+    await step('Character count for 0 characters remaining', async () => {
+      await userEvent.type(input, ' ');
+
+      expect(await canvas.findByText('Nog 0 karakters over.')).toBeVisible();
+      await waitFor(() => {
+        expect(input).toHaveAccessibleDescription('Nog 0 karakters over.');
+      });
+    });
+
+    await step('Character count beyond characters limit', async () => {
+      await userEvent.type(input, 'And I can go beyond it!');
+
+      expect(await canvas.findByText('Nog -23 karakters over.')).toBeVisible();
```

Comment:
Should we have another message if the text exceeds the limit? Without negative numbers

Reply:
I created #211 for this - keeping it out of scope for now to keep the port from SDK -> renderer as close as possible.